tests/transports/smpp/test_client.py

```diff
@@ -390,3 +392,32 @@ async def test_handle_deliver_sm_delivery_report(
     resp = await smsc.receive_pdu()
     assert isinstance(resp, DeliverSMResp)
     assert resp.seqNum == pdu.seqNum
+
+
+@pytest.mark.xfail(raises=SmscUnbind)
```

Comment:
Is this a test we're expecting to fail because it's a bug we haven't fixed or a feature we haven't added?
* If so, a `reason="why we expect this to fail"` arg is useful.
* If not, `with pytest.raises(...):` inside the test is a better option.

Reply:
So the issue is that, the exception gets raised by a child task running on the nursery, and I'm not sure how to get that to bubble up in the test method.

So `with pytest.raises(...):` doesn't catch it, because it only bubbles up when the test is being cleaned up.

Any ideas on how to get the exception to trigger? I guess I could call the `handle_pdu` or `handle_unbind` methods on the client directly instead of sending the PDU through the stream.

Reply:
Since nothing in `EsmeClient` uses its nursery until we call `.start()`, maybe we could replace that nursery with one that we create in the test? Something like:
```python
with pytest.raises(SmscUnbind):
    async with open_nursery() as nursery:
        client.nursery = nursery
        try:
            await smsc.start_and_bind(client)
            ...
        finally:
            nursery.cancel_scope.cancel()
```

If this is something we do more than once, it might be worth packing in a wrapper around `smsc.start_and_bind()` or something.

Reply:
Good idea, thanks, let me try that